### ISSUE TITLE
fields for group_by selection

### ISSUE THREAD

**Issue by Nello-Angelo** [VictoriaMetrics/VictoriaMetrics#7365]
### Is your feature request related to a problem? Please describe

yes

### Describe the solution you'd like

it would be nice to add a feature to combine these fields for group_by selection

**Comment by Nello-Angelo** [comment]
I would like to add that there is one inconvenient UI functionality - when I make a selection by group_by, the field by which I group shows me all the objects that belong to this field, for example "cluster" or "pod_name", and these objects can be there are many 20 clusters or hundreds of pods, but I would like to be able to group by individual fields to select objects individually, for example, the logs of cluster A and the logs of service B in cluster A. In data source loki there is such an opportunity

**Comment by valyala** [comment]
@Nello-Angelo , let me clarify the feature request, so we are on the same page. You want a quick way for narrowing down logs by applying filters for particular fields. For example, clicking `cluster:"foo"` label at any displayed log entry must add `cluster:"foo"` filter to the displayed logs. If this is the feature you want to have, then this feature should have the following requirements:

- The [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui) should contain a line of the currently applied `field:value` filters. The line must be located below the query input field.
- Every applied filter must be displayed as `field:value` followed by `x` button. When `x` button is clicked, then the corresponding filter must be dropped.
- There should be two filter types:
   1. Filters on [stream fields](https://docs.victoriametrics.com/victorialogs/keyconcepts/#stream-fields). These filters must be displayed as `{field:value}`. These filters can be selected by `ctrl`+clicking `field:value` inside `_stream` field.
   2. Filters on [other fields](https://docs.victoriametrics.com/victorialogs/keyconcepts/#other-fields). These filters must be displayed as `field:value`. These filters can be selected by `ctrl`+clicking  `field:value` inside any log entry.
- Web UI must pass all the selected filters to VictoriaLogs server via `extra_stream_filters` and `extra_filters` query args as JSON object. For example, if `cluster:foo` filter is selected, then it must be passed to VictoriaLogs as `extra_filters={"cluster":"foo"}`. If `pod:bar` additional filter is selected, then `extra_filters={"cluster":"foo","pod":"bar"}` must be passed to VictoriaLogs. The selected stream filters must be passed via `extra_stream_filters` query arg.

The support for `extra_stream_filters` and `extra_filters` query args has been added in the commit 7603446850778d770e6fe9f94c2306d7988ff6e5 . This commit will be included in the next release of VictoriaLogs. See [these docs](https://docs.victoriametrics.com/victorialogs/querying/#extra-filters) for details on how to use these query args.

@Loori-R , it would be great if you could implement such a quick filtering in VictoriaLogs web UI described above.

**Comment by Nello-Angelo** [comment]
@valyala yes in the current version I don't see an intuitive way to do what is described, I now tried to see a similar feature again but I didn't see or find the ability to combine key1:value1 key2:value2... in the left column of group_by. Maybe it's even worth changing the web design. By the way, maybe it's even worth making it so that those fields and their values ​​that I combined are additionally displayed as an expression in the search bar or simply as an annotation for what I clicked as a description. For example {cluster:"A" and pod:B}
In other words, the described feature is just a quick way to discard unnecessary things

**Comment by valyala** [comment]
@Loori-R , FYI, VictoriaLogs supports `extra_filters` and `extra_stream_filters` query args starting from [v0.40.0-victorialogs](https://docs.victoriametrics.com/victorialogs/changelog/). See [these docs](https://docs.victoriametrics.com/victorialogs/querying/#extra-filters) for details. So it is time to implement quick filtering at VictoriaLogs web UI as described [above](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7365#issuecomment-2447964259).

**Comment by tumd** [comment]
I'm a little bit confused about the new `extra_filters` & `extra_stream_filters` feature. It doesn't seem to fill the similar functionality that already exists in [VictoriaMetrics](https://docs.victoriametrics.com/#prometheus-querying-api-enhancements).

I was expecting/hoping these query args would fill [this feature request](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5542).

**Comment by valyala** [comment]
@tumd , thanks to pointing to the feature request https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5542 ! `extra_filters` and `extra_stream_filters` implement this feature.

**Comment by valyala** [comment]
@Loori-R , FYI, the next release of VictoriaLogs will support passing multiple values per each field in the `extra_filters` and `extra_stream_filters` query args according to [this comment](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5542#issuecomment-2536139341). This should allow selecting multiple values for the same log field in extra filters. This should allow creating faceted search with the recently added [`/select/logsql/facets` API](https://docs.victoriametrics.com/victorialogs/querying/#querying-facets).

**Comment by valyala** [comment]
FYI, VictoriaLogs supports multiple values per each field in the `extra_filters` and `extra_stream_filters` query args starting from [v1.4.0-victorialogs release](https://docs.victoriametrics.com/victorialogs/changelog/). See [these docs](https://docs.victoriametrics.com/victorialogs/querying/#extra-filters) for details.
